### PR TITLE
Feat/convert date

### PR DESCRIPTION
**Pull request by mansurishmuhametov**
## PR Checklist

Please check if your PR fulfills the following requirements:

-   [x] The commit message follows our guidelines: https://github.com/ngxs/store/blob/master/CONTRIBUTING.md#commit
-   [x] Tests for the changes have been added (for bug fixes / features)
-   [x] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

```
[ ] Bugfix
[x] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[ ] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Added date converters: shiftDate, startOfDay, endOfDay

Issue Number: N/A

## Does this PR introduce a breaking change?

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: packages/common/date/src/shift-date/shift-date.ts [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import { Interval } from './interval.interface';
+
+// eslint-disable-next-line complexity
+export function shiftDate(interval: Interval, date: Date = new Date()): Date {
+    const years: number = date.getFullYear() + (interval.years || 0);
```
Comment (Schirbak) [inline packages/common/date/src/shift-date/shift-date.ts:5]: не хочу мутабельную :( давай клонировать эту дату

File: packages/common/date/src/shift-date/interval.interface.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export interface Interval {
```
Comment (Schirbak) [inline packages/common/date/src/shift-date/interval.interface.ts:1]: давай переименуем в DateShiftDescriptor
@splincode як думаешь?

**Review by Schirbak — CHANGES_REQUESTED** [review]

File: packages/common/date/src/shift-date/interval.interface.ts (hunk 2 above)
Comment (splincode) [reply packages/common/date/src/shift-date/interval.interface.ts:1]: поддержива

File: packages/common/integration/tests/date/serial-date.spec.ts [hunk 3]
```diff
@@ -10,7 +10,7 @@ import {
 
 const isoFormat: string = 'yyyy-MM-dd HH:mm:ss';
 
-describe('[TEST]: Date', (): void => {
+xdescribe('[TEST]: Date', (): void => {
```
Comment (splincode) [inline packages/common/integration/tests/date/serial-date.spec.ts:13]: xdescribe?

**Review by splincode — DISMISSED** [review]

**Review by splincode — APPROVED** [review]

**Review by Schirbak — APPROVED** [review]

**Pull request merged**